scripts/pylib/twister/twisterlib/scripting.py

```diff
@@ -0,0 +1,96 @@
+            return matched_scripting
+        return None
+
+    def load_and_validate_files(self):
+        for scripting_file in self.scripting_files:
+            self.scripting.extend(ScriptingData.load_from_yaml(scripting_file, self.scripting_schema))
+
+@dataclass
+# Represents a single scripting element with associated scripts and metadata.
+class ScriptingElement:
+    scenarios: list[str] = field(default_factory=list)
+    platforms: list[str] = field(default_factory=list)
+    pre_script: str | None = None
+    post_flash_script: str | None = None
+    post_script: str | None = None
+    comment: str = 'NA'
+    re_scenarios: list[re.Pattern] = field(init=False, default_factory=list)
+    re_platforms: list[re.Pattern] = field(init=False, default_factory=list)
+
+    # Compiles regex patterns for scenarios and platforms, and validates the element.
+    def __post_init__(self):
+        self.re_scenarios = [re.compile(pat) for pat in self.scenarios if pat != 'all']
+        self.re_platforms = [re.compile(pat) for pat in self.platforms if pat != 'all']
+        if not any([self.scenarios, self.platforms, self.pre_script, self.post_flash_script, self.post_script]):
+            logger.error("At least one of the properties must be specified")
+            sys.exit(1)
+
+@dataclass
+# Holds a collection of scripting elements.
+class ScriptingData:
+    elements: list[ScriptingElement] = field(default_factory=list)
+
+    # Ensures all elements are ScriptingElement instances.
+    def __post_init__(self):
+        self.elements = [elem if isinstance(elem, ScriptingElement) else ScriptingElement(**elem) for elem in self.elements]
+
+    @classmethod
+    # Loads scripting data from a YAML file.
+    def load_from_yaml(cls, filename: Path | str, schema: dict) -> ScriptingData:
+        try:
+            raw_data = scl.yaml_load_verify(filename, schema) or []
+            return cls(raw_data)
+        except scl.EmptyYamlFileException:
+            logger.error(f'Scripting file {filename} is empty')
+            sys.exit(1)
+        except FileNotFoundError:
+            logger.error(f'Scripting file {filename} not found')
+            sys.exit(1)
+        except Exception as e:
+            logger.error(f'Error loading {filename}: {e}')
+            sys.exit(1)
+
+    # Extends the current scripting data with another set of scripting data.
+    def extend(self, other: ScriptingData) -> None:
+        self.elements.extend(other.elements)
+
+    # Finds a scripting element that matches the given scenario and platform.
+    def find_matching_scripting(self, scenario: str, platform: str) -> ScriptingElement | None:
+        for element in self.elements:
+            if element.scenarios and not _matches_element(scenario, element.re_scenarios):
+                continue
+            if element.platforms and not _matches_element(platform, element.re_platforms):
+                continue
+            return element
```

Comment:
wrt config duplicates, this feature differs from quarantine lookup and it might be needed to count all the matches first, choose one from them, and handle its duplicates with a warning.

Reply:
added extra parameter " override_script" which will